app/auth/routes.py

```diff
@@ -20,14 +21,68 @@ def login():
         if user is None or not user.check_password(form.password.data):
             flash(_('Invalid username or password'))
             return redirect(url_for('auth.login'))
-        login_user(user, remember=form.remember_me.data)
         next_page = request.args.get('next')
+        if user.two_factor_enabled():
+            request_verification_token(user.verification_phone)
+            session['username'] = user.username
+            session['phone'] = user.verification_phone
+            return redirect(url_for(
+                'auth.verify_2fa', next=next_page,
+                remember='1' if form.remember_me.data else '0'))
+        login_user(user, remember=form.remember_me.data)
         if not next_page or url_parse(next_page).netloc != '':
             next_page = url_for('main.index')
         return redirect(next_page)
     return render_template('auth/login.html', title=_('Sign In'), form=form)
 
 
+@bp.route('/verify2fa', methods=['GET', 'POST'])
+def verify_2fa():
```

Comment:
I cant find where this function is called. There are no requests to it.

Why does this function support both GET and POST? if its a form it should only be POST or PUT.

Reply:
The login function uses the verify_2fa function but the route itself isn't referenced anywhere directly. It's used to access the page after calling the function in redirect calls though.

Reply:
It gets the form's html template so I believe having a GET method is fine here and in other routes though I agree it may not be the best way to do things.